iphone/Classes/TiAppiOSUserActivityProxy.m

```diff
@@ -0,0 +1,446 @@
+/**
+ * Appcelerator Titanium Mobile
+ * Copyright (c) 2015 by Appcelerator, Inc. All Rights Reserved.
+ * Licensed under the terms of the Apache Public License
+ * Please see the LICENSE included with this distribution for details.
+ */
+
+#import "TiAppiOSUserActivityProxy.h"
+#import "TiUtils.h"
+
+#ifdef USE_TI_APPIOS
+
+@implementation TiAppiOSUserActivityProxy
+
+#pragma mark Titanium Proxy components
+
+-(NSString*)apiName
+{
+    return @"Ti.App.iOS.UserActivity";
+}
+
+-(id)initWithOptions:(NSDictionary*)props
+{
+    if (self = [super init]) {
+        _supported = [self determineMinRequirements:props];
+        if(_supported)
+        {
+            [self buildInitialActivity:props];
+        }
+    }
+    return self;
+}
+
+-(void)dealloc
+{
+    if(_supported)
+    {
+        [self clean];
+    }
+    [super dealloc];
+}
+
+#pragma mark internal helpers
+
+-(BOOL)activityTypeValid:(NSString*)activityType
+{
+    NSArray *supportedActivityTypes = [[NSBundle mainBundle]
+                                       objectForInfoDictionaryKey:@"NSUserActivityTypes"];
+    
+    return[supportedActivityTypes containsObject:activityType];
+}
+
+-(BOOL) determineMinRequirements:(NSDictionary *)props
+{
+    _isValid = NO;
+    if([TiUtils isIOS8OrGreater])
+    {
+        if([props objectForKey:@"activityType"])
+        {
+            NSArray *supportedActivityTypes = [[NSBundle mainBundle] objectForInfoDictionaryKey:@"NSUserActivityTypes"];
+            
+            if(![self activityTypeValid:[TiUtils stringValue:@"activityType" properties:props]])
+            {
+                NSLog(@"[ERROR] activityType provided is not defined in your projects tiapp.xml file");
+                return NO;
+            }
+            else
+            {
+                _isValid = YES;
+                return YES;
+            }
+        }
+        else
+        {
+            NSLog(@"[ERROR] activityType property is required on creation");
+            return NO;
+        }
+    }
+    
+    NSLog(@"[ERROR] %@ requires iOS8 or greater", [self apiName]);
+    return NO;
+}
+
+-(void) buildInitialActivity:(NSDictionary *)props
+{
+    NSString* activityType = [TiUtils stringValue:@"activityType" properties:props];
+    
+    [self instanceMake:activityType];
+    
+    if([props objectForKey:@"title"])
+    {
```

Comment:
For standardized coding convention, please use curly bracket immediately after if condition instead of next line throughout all the code. (i.e. `if([props objectForKey:@"title"]) {` )